lib/streamlit/connections/connection_factory.py

```diff
@@ -65,6 +68,13 @@ def connection(
     ...
 
 
+@overload
+def connection(
+    connection_class: Literal["file"], name: str = "default", **kwargs
+) -> "File":
```

Comment:
nit: I think this can be `File` (without the quotes) now that it's always imported rather than living in the `if TYPE_CHECKING` block (same thing with the `"Snowpark"` and `"SQL"` -- I just wrote those `@overloads` while the imports still lived in the if statement)